### PR TITLE
chore: ensure logs consistency across Coder

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/6726

This PR extends the `ruleguard` rules with `slog`. Actually, we caught a few incorrect or vague log messages.

New rules:
* Key: field `uuid.UUID` should have the key with `_id` suffix. Sample: `job` -> `job_id`
* Key: field should be snakecase
* Message: should start with lowercase and don't end with period
* Message: should contain at least 16 chars so that DevOps can set alarming
* Replace `slog.F("err", err)` with `slog.Error`

I'm fine with adding more rules if you have any ideas.

### CODE REVIEW

**Review by mafredri — APPROVED** [review]
Minor suggestions, but all in all looks good 👍🏻

File: agent/agent.go [hunk 1]
```diff
@@ -1154,7 +1154,7 @@ func (a *agent) handleReconnectingPTY(ctx context.Context, logger slog.Logger, m
 	err := rpty.ptty.Resize(msg.Height, msg.Width)
 	if err != nil {
 		// We can continue after this, it's not fatal!
-		logger.Error(ctx, "resize", slog.Error(err))
+		logger.Error(ctx, "reconnecting PTY failed resizing, but will continue", slog.Error(err))
```
[the remark this answers is not in the document]
Comment (mtojek) [reply agent/agent.go]: Fixed!

File: coderd/authorize.go [hunk 2]
```diff
@@ -71,11 +71,11 @@ func (h *HTTPAuthorizer) Authorize(r *http.Request, action rbac.Action, object r
 		internalError := new(rbac.UnauthorizedError)
 		logger := h.Logger
 		if xerrors.As(err, internalError) {
-			logger = h.Logger.With(slog.F("internal", internalError.Internal()))
+			logger = h.Logger.With(slog.F("internal_error", internalError.Internal()))
 		}
 		// Log information for debugging. This will be very helpful
 		// in the early days
-		logger.Warn(r.Context(), "unauthorized",
+		logger.Warn(r.Context(), "user is not authorized to access the object",
```
[the remark this answers is not in the document]
Comment (mtojek) [reply coderd/authorize.go]: s/user/requester/

File: cli/agent.go [hunk 3]
```diff
@@ -126,7 +126,7 @@ func (r *RootCmd) workspaceAgent() *clibase.Cmd {
 					reaper.WithCatchSignals(InterruptSignals...),
 				)
 				if err != nil {
-					logger.Error(ctx, "failed to reap", slog.Error(err))
+					logger.Error(ctx, "agent process reaper failed forking", slog.Error(err))
```
[the remark this answers is not in the document]
Comment (mtojek) [reply cli/agent.go]: Fixed

File: agent/agent.go [hunk 4]
```diff
@@ -1229,7 +1229,7 @@ func (a *agent) handleReconnectingPTY(ctx context.Context, logger slog.Logger, m
 		err = rpty.ptty.Resize(req.Height, req.Width)
 		if err != nil {
 			// We can continue after this, it's not fatal!
-			logger.Error(ctx, "resize", slog.Error(err))
+			logger.Error(ctx, "reconnecting PTY failed resizing", slog.Error(err))
```
[the remark this answers is not in the document]
Comment (mtojek) [reply agent/agent.go]: Fixed

**Pull request merged**